From 50b4e0b13d1b08945503fadd84ddf6e6d86b8d74 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Wed, 17 Nov 2021 11:40:04 +0100
Subject: [PATCH] Revert "[WFX-198] Support building with MinGW targeting
 Windows"

This reverts commit 4e9066418505de70c0d5316f7bb6624e3cf9cabc.

This breaks build on Linux
---
 browser/components/BrowserGlue.jsm            | 20 -------------------
 build/moz.configure/lto-pgo.configure         |  7 ++-----
 .../webrtc/libwebrtcglue/VideoConduit.cpp     |  4 ++--
 .../webrtc/sdp/RsdparsaSdpAttributeList.cpp   |  1 -
 dom/media/webrtc/sdp/moz.build                |  1 -
 .../transport/nrinterfaceprioritizer.cpp      |  6 +++---
 dom/media/webrtc/transport/sigslot.h          |  2 +-
 .../third_party/nICEr/src/net/local_addr.c    |  6 +++---
 .../third_party/nICEr/src/net/local_addr.h    |  7 +------
 .../nICEr/src/stun/addrs-netlink.c            | 16 +++++++--------
 .../third_party/nICEr/src/stun/addrs-win32.c  | 10 +++++-----
 .../third_party/nICEr/src/stun/addrs.c        |  2 +-
 .../third_party/nrappkit/src/log/r_log.c      |  2 +-
 .../nrappkit/src/registry/registry.c          |  2 +-
 gfx/angle/targets/libGLESv2/moz.build         |  3 ---
 netwerk/sctp/src/moz.build                    |  2 --
 netwerk/sctp/src/netinet/sctp_cc_functions.c  |  4 ----
 netwerk/sctp/src/netinet/sctp_indata.c        |  5 -----
 netwerk/sctp/src/netinet/sctp_input.c         |  4 ----
 netwerk/sctp/src/netinet/sctp_output.c        |  4 ----
 netwerk/sctp/src/netinet/sctp_usrreq.c        |  4 ----
 netwerk/sctp/src/netinet/sctputil.c           |  4 ----
 netwerk/sctp/src/user_mbuf.c                  |  4 ----
 netwerk/sctp/src/user_socket.c                |  4 ----
 .../webrtc/media/rtc_audio_video_gn/moz.build |  2 +-
 .../webrtc/media/rtc_media_gn/moz.build       |  2 +-
 .../audio_coding/test/TwoWayCommunication.cc  |  2 +-
 .../modules/desktop_capture/win/d3d_device.h  |  4 ++--
 .../win/desktop_device_info_win.cc            |  2 +-
 .../win/dxgi_adapter_duplicator.cc            |  2 +-
 .../win/dxgi_duplicator_controller.h          |  2 +-
 .../win/dxgi_output_duplicator.cc             |  6 +++---
 .../win/dxgi_output_duplicator.h              |  4 ++--
 .../desktop_capture/win/dxgi_texture.cc       |  2 +-
 .../desktop_capture/win/dxgi_texture.h        |  4 ++--
 .../win/dxgi_texture_mapping.cc               |  4 ++--
 .../win/dxgi_texture_mapping.h                |  4 ++--
 .../win/dxgi_texture_staging.cc               |  4 ++--
 .../win/dxgi_texture_staging.h                |  4 ++--
 .../win/screen_capturer_win_directx.h         |  2 +-
 .../win/screen_capturer_win_magnifier.cc      |  8 --------
 .../desktop_capture/window_capturer_win.cc    |  3 +--
 .../webrtc/modules/video_capture/BUILD.gn     |  2 +-
 .../video_capture_internal_impl_gn/moz.build  |  2 +-
 .../video_capture/windows/device_info_ds.cc   |  6 +-----
 .../webrtc/system_wrappers/source/clock.cc    |  2 +-
 .../system_wrappers/source/event_timer_win.cc |  2 +-
 .../system_wrappers/source/file_impl.cc       |  2 +-
 .../system_wrappers/source/rw_lock_win.h      |  2 +-
 .../libwebrtc/webrtc/test/win/d3d_renderer.h  |  2 +-
 .../libwebrtc/webrtc/test/win/run_loop_win.cc |  2 +-
 .../libwebrtc/webrtc/webrtc_gn/moz.build      |  2 +-
 third_party/sipcc/cpr_win_types.h             |  4 ----
 third_party/sipcc/moz.build                   |  1 -
 third_party/sipcc/sdp_token.c                 |  4 ----
 toolkit/content/gmp-sources/widevinecdm.json  |  3 ---
 56 files changed, 64 insertions(+), 157 deletions(-)

diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index b48a947b5ad0..29d4a903785a 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -2151,23 +2151,6 @@ BrowserGlue.prototype = {
     Services.wm.addListener(windowListener);
   },
 
-  _monitorEME() {
-    const PREF = "media.eme.enabled";
-    const _checkEMEPref = async () => {
-      let env = Cc["@mozilla.org/process/environment;1"].getService(
-        Ci.nsIEnvironment
-      );
-      let disabled = Services.prefs.getBoolPref(PREF, false);
-      if (disabled) {
-        env.set("MOZ_DISABLE_GMP_SANDBOX", "1");
-      } else {
-        env.set("MOZ_DISABLE_GMP_SANDBOX", "0");
-      }
-    };
-    Services.prefs.addObserver(PREF, _checkEMEPref);
-    _checkEMEPref();
-  },
-
   // All initial windows have opened.
   _onWindowsRestored: function BG__onWindowsRestored() {
     if (this._windowsWereRestored) {
@@ -2240,9 +2223,6 @@ BrowserGlue.prototype = {
     this._monitorHTTPSOnlyPref();
     this._monitorIonPref();
     this._monitorIonStudies();
-    if (AppConstants.platform == "win") {
-      this._monitorEME();
-    }
     if (AppConstants.NIGHTLY_BUILD) {
       this._monitorTranslationsPref();
     }
diff --git a/build/moz.configure/lto-pgo.configure b/build/moz.configure/lto-pgo.configure
index d922b9fd3a87..1fe5a1ab7375 100644
--- a/build/moz.configure/lto-pgo.configure
+++ b/build/moz.configure/lto-pgo.configure
@@ -292,7 +292,7 @@ def lto(
         # Tell LTO not to inline functions above a certain size, to mitigate
         # binary size growth while still getting good performance.
         # (For hot functions, PGO will put a multiplier on this limit.)
-        if target.os == "WINNT" and c_compiler.type == "clang-cl":
+        if target.os == "WINNT":
             ldflags.append("-mllvm:-import-instr-limit=10")
         elif target.os == "OSX":
             ldflags.append("-Wl,-mllvm,-import-instr-limit=10")
@@ -303,15 +303,12 @@ def lto(
         # during LTO. Further we can use the resulting size savings to increase
         # the import limit in hot functions.
         if newpm_flags:
-            if target.os == "WINNT" and c_compiler.type == "clang-cl":
+            if target.os == "WINNT":
                 # On Windows, this flag requires a change from clang-12, which
                 # is applied as a patch to our automation toolchain.
                 if automation or c_compiler.version >= "12.0.0":
                     ldflags.append("-opt:ltonewpassmanager")
                     ldflags.append("-mllvm:-import-hot-multiplier=30")
-            elif target.os == "WINNT" and c_compiler.type == "clang":
-                ldflags.append("-Wl,-plugin-opt=new-pass-manager")
-                ldflags.append("-Wl,-plugin-opt=-import-hot-multiplier=30")
             elif select_linker.KIND != "ld64" and c_compiler.type == "clang":
                 ldflags.append("-Wl,-plugin-opt=new-pass-manager")
                 ldflags.append("-Wl,-plugin-opt=-import-hot-multiplier=30")
diff --git a/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp b/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
index d205995499a8..18bcca75c5d3 100644
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
@@ -52,8 +52,8 @@
 #include "MediaDataCodec.h"
 
 // for ntohs
-#if (_MSC_VER || __MINGW32__)
-#  include "winsock2.h"
+#ifdef _MSC_VER
+#  include "Winsock2.h"
 #else
 #  include <netinet/in.h>
 #endif
diff --git a/dom/media/webrtc/sdp/RsdparsaSdpAttributeList.cpp b/dom/media/webrtc/sdp/RsdparsaSdpAttributeList.cpp
index 2eb2e3d2af9e..4e841610ebff 100644
--- a/dom/media/webrtc/sdp/RsdparsaSdpAttributeList.cpp
+++ b/dom/media/webrtc/sdp/RsdparsaSdpAttributeList.cpp
@@ -4,7 +4,6 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#include <string>
 #include "nsCRT.h"
 
 #include "sdp/RsdparsaSdpAttributeList.h"
diff --git a/dom/media/webrtc/sdp/moz.build b/dom/media/webrtc/sdp/moz.build
index 8494ea2b756f..675009acd07a 100644
--- a/dom/media/webrtc/sdp/moz.build
+++ b/dom/media/webrtc/sdp/moz.build
@@ -6,7 +6,6 @@
 
 if CONFIG["OS_TARGET"] == "WINNT":
     DEFINES["SIP_OS_WINDOWS"] = True
-    DEFINES['WEBRTC_MOZILLA_BUILD'] = True
 elif CONFIG["OS_TARGET"] == "Darwin":
     DEFINES["SIP_OS_OSX"] = True
 else:
diff --git a/dom/media/webrtc/transport/nrinterfaceprioritizer.cpp b/dom/media/webrtc/transport/nrinterfaceprioritizer.cpp
index a483fbe6f51a..9037b48c2024 100644
--- a/dom/media/webrtc/transport/nrinterfaceprioritizer.cpp
+++ b/dom/media/webrtc/transport/nrinterfaceprioritizer.cpp
@@ -44,9 +44,9 @@ class LocalAddress {
     }
     addr_ = buf;
 
-    is_vpn_ = (local_addr.Interface.type & NR_INTERFACE_TYPE_VPN) != 0 ? 1 : 0;
-    estimated_speed_ = local_addr.Interface.estimated_speed;
-    type_preference_ = GetNetworkTypePreference(local_addr.Interface.type);
+    is_vpn_ = (local_addr.interface.type & NR_INTERFACE_TYPE_VPN) != 0 ? 1 : 0;
+    estimated_speed_ = local_addr.interface.estimated_speed;
+    type_preference_ = GetNetworkTypePreference(local_addr.interface.type);
     ip_version_ = local_addr.addr.ip_version;
     return true;
   }
diff --git a/dom/media/webrtc/transport/sigslot.h b/dom/media/webrtc/transport/sigslot.h
index 0ca60ccf1bd4..448e8137fec7 100644
--- a/dom/media/webrtc/transport/sigslot.h
+++ b/dom/media/webrtc/transport/sigslot.h
@@ -114,7 +114,7 @@
 #    define WIN32_LEAN_AND_MEAN
 #  endif
 #  include "rtc_base/win32.h"
-#elif (defined(__GNUG__) || defined(SIGSLOT_USE_POSIX_THREADS)) && !defined(__MINGW32__)
+#elif defined(__GNUG__) || defined(SIGSLOT_USE_POSIX_THREADS)
 #  define _SIGSLOT_HAS_POSIX_THREADS
 #  include <pthread.h>
 #else
diff --git a/dom/media/webrtc/transport/third_party/nICEr/src/net/local_addr.c b/dom/media/webrtc/transport/third_party/nICEr/src/net/local_addr.c
index 83ad249aac8f..a0896d5e9d8b 100644
--- a/dom/media/webrtc/transport/third_party/nICEr/src/net/local_addr.c
+++ b/dom/media/webrtc/transport/third_party/nICEr/src/net/local_addr.c
@@ -44,7 +44,7 @@ int nr_local_addr_copy(nr_local_addr *to, nr_local_addr *from)
     if (r=nr_transport_addr_copy(&(to->addr), &(from->addr))) {
       ABORT(r);
     }
-    to->Interface = from->Interface;
+    to->interface = from->interface;
     to->flags = from->flags;
 
     _status=0;
@@ -54,7 +54,7 @@ int nr_local_addr_copy(nr_local_addr *to, nr_local_addr *from)
 
 int nr_local_addr_fmt_info_string(nr_local_addr *addr, char *buf, int len)
   {
-    int addr_type = addr->Interface.type;
+    int addr_type = addr->interface.type;
     const char *vpn = (addr_type & NR_INTERFACE_TYPE_VPN) ? "VPN on " : "";
 
     const char *type = (addr_type & NR_INTERFACE_TYPE_WIRED) ? "wired" :
@@ -63,7 +63,7 @@ int nr_local_addr_fmt_info_string(nr_local_addr *addr, char *buf, int len)
                        "unknown";
 
     snprintf(buf, len, "%s%s, estimated speed: %d kbps %s",
-             vpn, type, addr->Interface.estimated_speed,
+             vpn, type, addr->interface.estimated_speed,
              (addr->flags & NR_ADDR_FLAG_TEMPORARY ? "temporary" : ""));
     buf[len - 1] = '\0';
     return (0);
diff --git a/dom/media/webrtc/transport/third_party/nICEr/src/net/local_addr.h b/dom/media/webrtc/transport/third_party/nICEr/src/net/local_addr.h
index 5f6613df6c28..058f7a5e6b1a 100644
--- a/dom/media/webrtc/transport/third_party/nICEr/src/net/local_addr.h
+++ b/dom/media/webrtc/transport/third_party/nICEr/src/net/local_addr.h
@@ -49,14 +49,9 @@ typedef struct nr_interface_ {
   int estimated_speed; /* Speed in kbps */
 } nr_interface;
 
-typedef struct nr_innrr_interface_ {
-  int type;
-  int estimated_speed; /* Speed in kbps */
-} nrr_interface;
-
 typedef struct nr_local_addr_ {
   nr_transport_addr addr;
-  nr_interface Interface;
+  nr_interface interface;
 #define NR_ADDR_FLAG_TEMPORARY 0x1
   int flags;
 } nr_local_addr;
diff --git a/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-netlink.c b/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-netlink.c
index ad190ee3e800..73e85c6cccad 100644
--- a/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-netlink.c
+++ b/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-netlink.c
@@ -139,7 +139,7 @@ stun_convert_netlink(nr_local_addr *addr, struct ifaddrmsg *address_msg, struct
   int flags = get_siocgifflags(addr);
   if (flags & IFF_POINTOPOINT)
   {
-    addr->Interface.type = NR_INTERFACE_TYPE_UNKNOWN | NR_INTERFACE_TYPE_VPN;
+    addr->interface.type = NR_INTERFACE_TYPE_UNKNOWN | NR_INTERFACE_TYPE_VPN;
     /* TODO (Bug 896913): find backend network type of this VPN */
   }
 
@@ -162,11 +162,11 @@ stun_convert_netlink(nr_local_addr *addr, struct ifaddrmsg *address_msg, struct
   {
     /* For wireless network, we won't get ethtool, it's a wired
      * connection */
-    addr->Interface.type = NR_INTERFACE_TYPE_WIRED;
+    addr->interface.type = NR_INTERFACE_TYPE_WIRED;
 #ifdef DONT_HAVE_ETHTOOL_SPEED_HI
-    addr->Interface.estimated_speed = ecmd.speed;
+    addr->interface.estimated_speed = ecmd.speed;
 #else
-    addr->Interface.estimated_speed = ((ecmd.speed_hi << 16) | ecmd.speed) * 1000;
+    addr->interface.estimated_speed = ((ecmd.speed_hi << 16) | ecmd.speed) * 1000;
 #endif
   }
 
@@ -174,15 +174,15 @@ stun_convert_netlink(nr_local_addr *addr, struct ifaddrmsg *address_msg, struct
   e = ioctl(s, SIOCGIWRATE, &wrq);
   if (e == 0)
   {
-    addr->Interface.type = NR_INTERFACE_TYPE_WIFI;
-    addr->Interface.estimated_speed = wrq.u.bitrate.value / 1000;
+    addr->interface.type = NR_INTERFACE_TYPE_WIFI;
+    addr->interface.estimated_speed = wrq.u.bitrate.value / 1000;
   }
 
   close(s);
 
 #else
-  addr->Interface.type = NR_INTERFACE_TYPE_UNKNOWN;
-  addr->Interface.estimated_speed = 0;
+  addr->interface.type = NR_INTERFACE_TYPE_UNKNOWN;
+  addr->interface.estimated_speed = 0;
 #endif
   return 0;
 }
diff --git a/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-win32.c b/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-win32.c
index 1a7a6f9d276b..4fdf5e5d4451 100644
--- a/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-win32.c
+++ b/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs-win32.c
@@ -174,18 +174,18 @@ stun_getaddrs_filtered(nr_local_addr addrs[], int maxaddrs, int *count)
 
           strlcpy(addrs[n].addr.ifname, hex_hashed_ifname, sizeof(addrs[n].addr.ifname));
           if (tmpAddress->IfType == IF_TYPE_ETHERNET_CSMACD) {
-            addrs[n].Interface.type = NR_INTERFACE_TYPE_WIRED;
+            addrs[n].interface.type = NR_INTERFACE_TYPE_WIRED;
           } else if (tmpAddress->IfType == IF_TYPE_IEEE80211) {
             /* Note: this only works for >= Win Vista */
-            addrs[n].Interface.type = NR_INTERFACE_TYPE_WIFI;
+            addrs[n].interface.type = NR_INTERFACE_TYPE_WIFI;
           } else {
-            addrs[n].Interface.type = NR_INTERFACE_TYPE_UNKNOWN;
+            addrs[n].interface.type = NR_INTERFACE_TYPE_UNKNOWN;
           }
 #if (_WIN32_WINNT >= 0x0600)
           /* Note: only >= Vista provide link speed information */
-          addrs[n].Interface.estimated_speed = tmpAddress->TransmitLinkSpeed / 1000;
+          addrs[n].interface.estimated_speed = tmpAddress->TransmitLinkSpeed / 1000;
 #else
-          addrs[n].Interface.estimated_speed = 0;
+          addrs[n].interface.estimated_speed = 0;
 #endif
           if (stun_win32_address_temp_v6(u)) {
             addrs[n].flags |= NR_ADDR_FLAG_TEMPORARY;
diff --git a/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs.c b/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs.c
index 70dafa297c8e..362b7d828e49 100644
--- a/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs.c
+++ b/dom/media/webrtc/transport/third_party/nICEr/src/stun/addrs.c
@@ -82,7 +82,7 @@ nr_stun_filter_addrs(nr_local_addr addrs[], int remove_loopback, int remove_link
     for (i = 0; i < *count; ++i) {
       if (addrs[i].addr.ip_version == NR_IPV6) {
         if (nr_transport_addr_is_teredo(&addrs[i].addr)) {
-            addrs[i].Interface.type |= NR_INTERFACE_TYPE_TEREDO;
+            addrs[i].interface.type |= NR_INTERFACE_TYPE_TEREDO;
             /* Prefer teredo over mac-based address. Probably will never see
              * both. */
             filter_mac_ipv6 = 1;
diff --git a/dom/media/webrtc/transport/third_party/nrappkit/src/log/r_log.c b/dom/media/webrtc/transport/third_party/nrappkit/src/log/r_log.c
index 1371103933a6..09bb24749fca 100644
--- a/dom/media/webrtc/transport/third_party/nrappkit/src/log/r_log.c
+++ b/dom/media/webrtc/transport/third_party/nrappkit/src/log/r_log.c
@@ -46,7 +46,7 @@
 
 #include <string.h>
 #include <errno.h>
-#if !(defined(_MSC_VER) || defined(__MINGW32__))
+#ifndef _MSC_VER
 #include <strings.h>
 #include <syslog.h>
 #endif
diff --git a/dom/media/webrtc/transport/third_party/nrappkit/src/registry/registry.c b/dom/media/webrtc/transport/third_party/nrappkit/src/registry/registry.c
index 3c21e5bb125b..709b1c3fb739 100644
--- a/dom/media/webrtc/transport/third_party/nrappkit/src/registry/registry.c
+++ b/dom/media/webrtc/transport/third_party/nrappkit/src/registry/registry.c
@@ -44,7 +44,7 @@
 
 #include <assert.h>
 #include <string.h>
-#if !(defined(_MSC_VER) || defined(__MINGW32__))
+#ifndef _MSC_VER
 #include <strings.h>
 #include <sys/param.h>
 #include <netinet/in.h>
diff --git a/gfx/angle/targets/libGLESv2/moz.build b/gfx/angle/targets/libGLESv2/moz.build
index 2f2f1dd2e8ce..ebcf0dbf2f99 100644
--- a/gfx/angle/targets/libGLESv2/moz.build
+++ b/gfx/angle/targets/libGLESv2/moz.build
@@ -436,9 +436,6 @@ OS_LIBS += [
     "ws2_32",
 ]
 
-if CONFIG["OS_ARCH"] == "WINNT" and CONFIG["CC_TYPE"] == "clang":
-    CXXFLAGS += ["-fno-lto"]
-
 # LDFLAGS += [
 #     "--color-diagnostics",
 #     "/call-graph-profile-sort:no",
diff --git a/netwerk/sctp/src/moz.build b/netwerk/sctp/src/moz.build
index 3833a195ef24..45d4e82ad201 100644
--- a/netwerk/sctp/src/moz.build
+++ b/netwerk/sctp/src/moz.build
@@ -56,5 +56,3 @@ if CONFIG['OS_TARGET'] in ('Linux', 'Android'):
     DEFINES['_GNU_SOURCE'] = 1
 elif CONFIG['OS_TARGET'] == 'Darwin':
     DEFINES['__APPLE_USE_RFC_2292'] = 1
-elif CONFIG['OS_TARGET'] == 'WINNT':
-    DEFINES['WEBRTC_MOZILLA_BUILD'] = 1
diff --git a/netwerk/sctp/src/netinet/sctp_cc_functions.c b/netwerk/sctp/src/netinet/sctp_cc_functions.c
index 1dafdaee8dde..dcf6298579a0 100755
--- a/netwerk/sctp/src/netinet/sctp_cc_functions.c
+++ b/netwerk/sctp/src/netinet/sctp_cc_functions.c
@@ -54,10 +54,6 @@ __FBSDID("$FreeBSD: head/sys/netinet/sctp_cc_functions.c 359405 2020-03-28 20:25
 #include <netinet/sctp_kdtrace.h>
 #endif
 
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 #define SHIFT_MPTCP_MULTI_N 40
 #define SHIFT_MPTCP_MULTI_Z 16
 #define SHIFT_MPTCP_MULTI 8
diff --git a/netwerk/sctp/src/netinet/sctp_indata.c b/netwerk/sctp/src/netinet/sctp_indata.c
index f87706a60239..256bd9ab0f5c 100755
--- a/netwerk/sctp/src/netinet/sctp_indata.c
+++ b/netwerk/sctp/src/netinet/sctp_indata.c
@@ -58,11 +58,6 @@ __FBSDID("$FreeBSD: head/sys/netinet/sctp_indata.c 363076 2020-07-10 11:15:10Z t
 #if defined(__FreeBSD__) && !defined(__Userspace__)
 #include <netinet/sctp_lock_bsd.h>
 #endif
-
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 /*
  * NOTES: On the outbound side of things I need to check the sack timer to
  * see if I should generate a sack into the chunk queue (if I have data to
diff --git a/netwerk/sctp/src/netinet/sctp_input.c b/netwerk/sctp/src/netinet/sctp_input.c
index 8b89adbed8cb..46c196002e31 100755
--- a/netwerk/sctp/src/netinet/sctp_input.c
+++ b/netwerk/sctp/src/netinet/sctp_input.c
@@ -63,10 +63,6 @@ __FBSDID("$FreeBSD: head/sys/netinet/sctp_input.c 362153 2020-06-13 18:38:59Z tu
 #include <sys/smp.h>
 #endif
 
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 static void
 sctp_stop_all_cookie_timers(struct sctp_tcb *stcb)
 {
diff --git a/netwerk/sctp/src/netinet/sctp_output.c b/netwerk/sctp/src/netinet/sctp_output.c
index 9d5f019c9ad2..3bfd2ef41439 100755
--- a/netwerk/sctp/src/netinet/sctp_output.c
+++ b/netwerk/sctp/src/netinet/sctp_output.c
@@ -80,10 +80,6 @@ __FBSDID("$FreeBSD: head/sys/netinet/sctp_output.c 362178 2020-06-14 16:05:08Z t
 #include <netinet6/sctp6_var.h>
 #endif
 
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 #if defined(__APPLE__) && !defined(__Userspace__)
 #if !(defined(APPLE_LEOPARD) || defined(APPLE_SNOWLEOPARD))
 #define SCTP_MAX_LINKHDR 16
diff --git a/netwerk/sctp/src/netinet/sctp_usrreq.c b/netwerk/sctp/src/netinet/sctp_usrreq.c
index 2ff358a2f91c..78286988f8b4 100755
--- a/netwerk/sctp/src/netinet/sctp_usrreq.c
+++ b/netwerk/sctp/src/netinet/sctp_usrreq.c
@@ -69,10 +69,6 @@ __FBSDID("$FreeBSD: head/sys/netinet/sctp_usrreq.c 362563 2020-06-23 23:05:05Z t
 #include <netinet/sctp_peeloff.h>
 #endif				/* HAVE_SCTP_PEELOFF_SOCKOPT */
 
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 extern const struct sctp_cc_functions sctp_cc_functions[];
 extern const struct sctp_ss_functions sctp_ss_functions[];
 
diff --git a/netwerk/sctp/src/netinet/sctputil.c b/netwerk/sctp/src/netinet/sctputil.c
index c33457f5e89e..e40e04e5914b 100755
--- a/netwerk/sctp/src/netinet/sctputil.c
+++ b/netwerk/sctp/src/netinet/sctputil.c
@@ -82,10 +82,6 @@ __FBSDID("$FreeBSD: head/sys/netinet/sctputil.c 362462 2020-06-21 09:56:09Z tuex
 #endif
 #endif
 
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 extern const struct sctp_cc_functions sctp_cc_functions[];
 extern const struct sctp_ss_functions sctp_ss_functions[];
 
diff --git a/netwerk/sctp/src/user_mbuf.c b/netwerk/sctp/src/user_mbuf.c
index 503936558d96..9e566e302590 100755
--- a/netwerk/sctp/src/user_mbuf.c
+++ b/netwerk/sctp/src/user_mbuf.c
@@ -46,10 +46,6 @@
 #include "user_atomic.h"
 #include "netinet/sctp_pcb.h"
 
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 #define KIPC_MAX_LINKHDR        4       /* int: max length of link header (see sys/sysclt.h) */
 #define KIPC_MAX_PROTOHDR	5	/* int: max length of network header (see sys/sysclt.h)*/
 int max_linkhdr = KIPC_MAX_LINKHDR;
diff --git a/netwerk/sctp/src/user_socket.c b/netwerk/sctp/src/user_socket.c
index cc92b549f4ef..b72b05ebc303 100755
--- a/netwerk/sctp/src/user_socket.c
+++ b/netwerk/sctp/src/user_socket.c
@@ -65,10 +65,6 @@ userland_cond_t accept_cond;
 #include <sys/timeb.h>
 #endif
 
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 MALLOC_DEFINE(M_PCB, "sctp_pcb", "sctp pcb");
 MALLOC_DEFINE(M_SONAME, "sctp_soname", "sctp soname");
 #define MAXLEN_MBUF_CHAIN  32
diff --git a/third_party/libwebrtc/webrtc/media/rtc_audio_video_gn/moz.build b/third_party/libwebrtc/webrtc/media/rtc_audio_video_gn/moz.build
index 281aadd04b70..4f6d5f5c0a8c 100644
--- a/third_party/libwebrtc/webrtc/media/rtc_audio_video_gn/moz.build
+++ b/third_party/libwebrtc/webrtc/media/rtc_audio_video_gn/moz.build
@@ -169,7 +169,7 @@ if CONFIG["OS_TARGET"] == "WINNT":
     DEFINES["__STD_C"] = True
 
     OS_LIBS += [
-        "strmiids",
+        "Strmiids",
         "winmm"
     ]
 
diff --git a/third_party/libwebrtc/webrtc/media/rtc_media_gn/moz.build b/third_party/libwebrtc/webrtc/media/rtc_media_gn/moz.build
index 36ed261b6a58..a8cbafc3a1d1 100644
--- a/third_party/libwebrtc/webrtc/media/rtc_media_gn/moz.build
+++ b/third_party/libwebrtc/webrtc/media/rtc_media_gn/moz.build
@@ -140,7 +140,7 @@ if CONFIG["OS_TARGET"] == "WINNT":
     DEFINES["__STD_C"] = True
 
     OS_LIBS += [
-        "strmiids",
+        "Strmiids",
         "winmm"
     ]
 
diff --git a/third_party/libwebrtc/webrtc/modules/audio_coding/test/TwoWayCommunication.cc b/third_party/libwebrtc/webrtc/modules/audio_coding/test/TwoWayCommunication.cc
index 68fea2f238de..8049436cd01f 100644
--- a/third_party/libwebrtc/webrtc/modules/audio_coding/test/TwoWayCommunication.cc
+++ b/third_party/libwebrtc/webrtc/modules/audio_coding/test/TwoWayCommunication.cc
@@ -17,7 +17,7 @@
 #include <memory>
 
 #ifdef WIN32
-#include <windows.h>
+#include <Windows.h>
 #endif
 
 #include "api/audio_codecs/builtin_audio_decoder_factory.h"
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/d3d_device.h b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/d3d_device.h
index 4ee003e0f3ec..8f545d367c5e 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/d3d_device.h
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/d3d_device.h
@@ -13,8 +13,8 @@
 
 #include <comdef.h>
 #include <wrl/client.h>
-#include <d3d11.h>
-#include <dxgi.h>
+#include <D3D11.h>
+#include <DXGI.h>
 
 #include <vector>
 
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/desktop_device_info_win.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/desktop_device_info_win.cc
index 76e5372b2224..45aa9a6484d1 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/desktop_device_info_win.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/desktop_device_info_win.cc
@@ -7,7 +7,7 @@
 #include "modules/desktop_capture/win/win_shared.h"
 #include <inttypes.h>
 #include <stdio.h>
-#include <versionhelpers.h>
+#include <VersionHelpers.h>
 
 // Duplicating declaration so that it always resolves in decltype use
 // typedef BOOL (WINAPI *QueryFullProcessImageNameProc)(HANDLE hProcess, DWORD dwFlags, LPTSTR lpExeName, PDWORD lpdwSize);
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_adapter_duplicator.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_adapter_duplicator.cc
index e3f11ac30af2..c4cd982d27cd 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_adapter_duplicator.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_adapter_duplicator.cc
@@ -11,7 +11,7 @@
 #include "modules/desktop_capture/win/dxgi_adapter_duplicator.h"
 
 #include <comdef.h>
-#include <dxgi.h>
+#include <DXGI.h>
 
 #include <algorithm>
 
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_duplicator_controller.h b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_duplicator_controller.h
index 2e390c69fbd4..ecae481005fd 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_duplicator_controller.h
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_duplicator_controller.h
@@ -11,7 +11,7 @@
 #ifndef MODULES_DESKTOP_CAPTURE_WIN_DXGI_DUPLICATOR_CONTROLLER_H_
 #define MODULES_DESKTOP_CAPTURE_WIN_DXGI_DUPLICATOR_CONTROLLER_H_
 
-#include <d3dcommon.h>
+#include <D3DCommon.h>
 
 #include <atomic>
 #include <string>
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_output_duplicator.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_output_duplicator.cc
index f5cbf8b392ab..b920821dea4c 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_output_duplicator.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_output_duplicator.cc
@@ -13,9 +13,9 @@
 #include <string.h>
 
 #include <unknwn.h>
-#include <dxgi.h>
-#include <dxgiformat.h>
-#include <windows.h>
+#include <DXGI.h>
+#include <DXGIFormat.h>
+#include <Windows.h>
 
 #include <algorithm>
 
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_output_duplicator.h b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_output_duplicator.h
index cbba1d6dc104..3f10533a2243 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_output_duplicator.h
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_output_duplicator.h
@@ -13,8 +13,8 @@
 
 #include <comdef.h>
 #include <wrl/client.h>
-#include <dxgi.h>
-#include <dxgi1_2.h>
+#include <DXGI.h>
+#include <DXGI1_2.h>
 
 #include <memory>
 #include <string>
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture.cc
index afc893e11b91..8f1d034a06bf 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture.cc
@@ -12,7 +12,7 @@
 
 #include <comdef.h>
 #include <wrl/client.h>
-#include <d3d11.h>
+#include <D3D11.h>
 
 #include "modules/desktop_capture/desktop_region.h"
 #include "rtc_base/checks.h"
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture.h b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture.h
index 208e6a9523b5..18fa349f820c 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture.h
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture.h
@@ -11,8 +11,8 @@
 #ifndef MODULES_DESKTOP_CAPTURE_WIN_DXGI_TEXTURE_H_
 #define MODULES_DESKTOP_CAPTURE_WIN_DXGI_TEXTURE_H_
 
-#include <d3d11.h>
-#include <dxgi1_2.h>
+#include <D3D11.h>
+#include <DXGI1_2.h>
 
 #include <memory>
 
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_mapping.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_mapping.cc
index 9e138d1d6fc8..dfe8259dd055 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_mapping.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_mapping.cc
@@ -11,8 +11,8 @@
 #include "modules/desktop_capture/win/dxgi_texture_mapping.h"
 
 #include <comdef.h>
-#include <dxgi.h>
-#include <dxgi1_2.h>
+#include <DXGI.h>
+#include <DXGI1_2.h>
 
 #include "rtc_base/checks.h"
 #include "rtc_base/logging.h"
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_mapping.h b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_mapping.h
index 043ce612a1a4..1d18a57131b1 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_mapping.h
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_mapping.h
@@ -11,8 +11,8 @@
 #ifndef MODULES_DESKTOP_CAPTURE_WIN_DXGI_TEXTURE_MAPPING_H_
 #define MODULES_DESKTOP_CAPTURE_WIN_DXGI_TEXTURE_MAPPING_H_
 
-#include <d3d11.h>
-#include <dxgi1_2.h>
+#include <D3D11.h>
+#include <DXGI1_2.h>
 
 #include "modules/desktop_capture/desktop_geometry.h"
 #include "modules/desktop_capture/desktop_region.h"
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_staging.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_staging.cc
index 057fcdadd410..e3e915726efb 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_staging.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_staging.cc
@@ -12,8 +12,8 @@
 
 #include <comdef.h>
 #include <unknwn.h>
-#include <dxgi.h>
-#include <dxgi1_2.h>
+#include <DXGI.h>
+#include <DXGI1_2.h>
 
 #include "rtc_base/checks.h"
 #include "rtc_base/logging.h"
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_staging.h b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_staging.h
index c505295d9920..65d56baed290 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_staging.h
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/dxgi_texture_staging.h
@@ -12,8 +12,8 @@
 #define MODULES_DESKTOP_CAPTURE_WIN_DXGI_TEXTURE_STAGING_H_
 
 #include <wrl/client.h>
-#include <d3d11.h>
-#include <dxgi1_2.h>
+#include <D3D11.h>
+#include <DXGI1_2.h>
 
 #include "modules/desktop_capture/desktop_geometry.h"
 #include "modules/desktop_capture/desktop_region.h"
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/screen_capturer_win_directx.h b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/screen_capturer_win_directx.h
index ed7083df093e..9438bbd0cf2a 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/screen_capturer_win_directx.h
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/screen_capturer_win_directx.h
@@ -11,7 +11,7 @@
 #ifndef MODULES_DESKTOP_CAPTURE_WIN_SCREEN_CAPTURER_WIN_DIRECTX_H_
 #define MODULES_DESKTOP_CAPTURE_WIN_SCREEN_CAPTURER_WIN_DIRECTX_H_
 
-#include <d3dcommon.h>
+#include <D3DCommon.h>
 
 #include <memory>
 #include <vector>
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/screen_capturer_win_magnifier.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/screen_capturer_win_magnifier.cc
index 39fbfc409292..df9953a1f83e 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/win/screen_capturer_win_magnifier.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/win/screen_capturer_win_magnifier.cc
@@ -24,12 +24,6 @@
 #include "rtc_base/logging.h"
 #include "rtc_base/timeutils.h"
 
-/*
-#define INITGUID
-#include <guiddef.h>
-DEFINE_GUID(GUID_WICPixelFormat32bppRGBA, 0xf5c7ad2d, 0x6a8d, 0x43dd, 0xa7, 0xa8, 0xa2, 0x99, 0x35, 0x26, 0x1a, 0xe9);
-*/
-
 namespace webrtc {
 
 namespace {
@@ -337,7 +331,6 @@ bool ScreenCapturerWinMagnifier::InitializeMagnifier() {
 
 void ScreenCapturerWinMagnifier::OnCaptured(void* data,
                                             const MAGIMAGEHEADER& header) {
-  #if 0
   DesktopFrame* current_frame = queue_.current_frame();
 
   // Verify the format.
@@ -366,7 +359,6 @@ void ScreenCapturerWinMagnifier::OnCaptured(void* data,
       DesktopRect::MakeXYWH(0, 0, header.width, header.height));
 
   magnifier_capture_succeeded_ = true;
-  #endif
 }
 
 void ScreenCapturerWinMagnifier::CreateCurrentFrameIfNecessary(
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/window_capturer_win.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/window_capturer_win.cc
index d7898d2e280e..9c659868e340 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/window_capturer_win.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/window_capturer_win.cc
@@ -21,7 +21,7 @@
 #include "rtc_base/constructormagic.h"
 #include "rtc_base/logging.h"
 #include "rtc_base/win32.h"
-#include <versionhelpers.h>
+#include <VersionHelpers.h>
 
 namespace webrtc {
 
@@ -317,7 +317,6 @@ void WindowCapturerWin::CaptureFrame() {
     // Windows 8.0 this flag is not supported so the block below will fallback
     // to the other call to PrintWindow. It seems to be very tricky to detect
     // Windows 8.0 vs 8.1 so a try/fallback is more approriate here.
-    #define PW_RENDERFULLCONTENT    0x00000002 //tma
     const UINT flags = PW_RENDERFULLCONTENT;
     result = PrintWindow(window_, mem_dc, flags);
   }
diff --git a/third_party/libwebrtc/webrtc/modules/video_capture/BUILD.gn b/third_party/libwebrtc/webrtc/modules/video_capture/BUILD.gn
index cccbaa13db09..6eef18a4b5d2 100644
--- a/third_party/libwebrtc/webrtc/modules/video_capture/BUILD.gn
+++ b/third_party/libwebrtc/webrtc/modules/video_capture/BUILD.gn
@@ -121,7 +121,7 @@ if (!build_with_chromium) {
         "windows/video_capture_factory_windows.cc",
       ]
 
-      libs = [ "strmiids.lib" ]
+      libs = [ "Strmiids.lib" ]
 
     }
 
diff --git a/third_party/libwebrtc/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build b/third_party/libwebrtc/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
index c255ebfb6059..4454a655a066 100644
--- a/third_party/libwebrtc/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
+++ b/third_party/libwebrtc/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
@@ -143,7 +143,7 @@ if CONFIG["OS_TARGET"] == "WINNT":
     DEFINES["__STD_C"] = True
 
     OS_LIBS += [
-        "strmiids",
+        "Strmiids",
         "winmm"
     ]
 
diff --git a/third_party/libwebrtc/webrtc/modules/video_capture/windows/device_info_ds.cc b/third_party/libwebrtc/webrtc/modules/video_capture/windows/device_info_ds.cc
index 42d5ff8076be..e48c438ffd6b 100644
--- a/third_party/libwebrtc/webrtc/modules/video_capture/windows/device_info_ds.cc
+++ b/third_party/libwebrtc/webrtc/modules/video_capture/windows/device_info_ds.cc
@@ -8,10 +8,6 @@
  *  be found in the AUTHORS file in the root of the source tree.
  */
 
-#define INITGUID
-#include <guiddef.h>
-DEFINE_GUID(KSCATEGORY_VIDEO_CAMERA, 0xe5323777, 0xf976, 0x4f5b, 0x9b, 0x55, 0xb9, 0x46, 0x99, 0xc4, 0x6e, 0x44);
-
 #include "modules/video_capture/windows/device_info_ds.h"
 
 #include "modules/video_capture/video_capture_config.h"
@@ -20,7 +16,7 @@ DEFINE_GUID(KSCATEGORY_VIDEO_CAMERA, 0xe5323777, 0xf976, 0x4f5b, 0x9b, 0x55, 0xb
 #include "rtc_base/stringutils.h"
 
 #include <dvdmedia.h>
-#include <dbt.h>
+#include <Dbt.h>
 #include <ks.h>
 #include <ksmedia.h>
 
diff --git a/third_party/libwebrtc/webrtc/system_wrappers/source/clock.cc b/third_party/libwebrtc/webrtc/system_wrappers/source/clock.cc
index 1d5480f95418..61fc4a49a6ce 100644
--- a/third_party/libwebrtc/webrtc/system_wrappers/source/clock.cc
+++ b/third_party/libwebrtc/webrtc/system_wrappers/source/clock.cc
@@ -15,7 +15,7 @@
 // Windows needs to be included before mmsystem.h
 #include "rtc_base/win32.h"
 
-#include <mmsystem.h>
+#include <MMSystem.h>
 
 #elif ((defined WEBRTC_LINUX) || (defined WEBRTC_MAC) || (defined WEBRTC_BSD))
 
diff --git a/third_party/libwebrtc/webrtc/system_wrappers/source/event_timer_win.cc b/third_party/libwebrtc/webrtc/system_wrappers/source/event_timer_win.cc
index c2ad9f372a28..b6a93feb8229 100644
--- a/third_party/libwebrtc/webrtc/system_wrappers/source/event_timer_win.cc
+++ b/third_party/libwebrtc/webrtc/system_wrappers/source/event_timer_win.cc
@@ -10,7 +10,7 @@
 
 #include "system_wrappers/source/event_timer_win.h"
 
-#include "mmsystem.h"
+#include "Mmsystem.h"
 
 namespace webrtc {
 
diff --git a/third_party/libwebrtc/webrtc/system_wrappers/source/file_impl.cc b/third_party/libwebrtc/webrtc/system_wrappers/source/file_impl.cc
index 9ba4ca48ca79..350aaeb15bd8 100644
--- a/third_party/libwebrtc/webrtc/system_wrappers/source/file_impl.cc
+++ b/third_party/libwebrtc/webrtc/system_wrappers/source/file_impl.cc
@@ -11,7 +11,7 @@
 #include "system_wrappers/include/file_wrapper.h"
 
 #ifdef _WIN32
-#include <windows.h>
+#include <Windows.h>
 #else
 #include <stdarg.h>
 #include <string.h>
diff --git a/third_party/libwebrtc/webrtc/system_wrappers/source/rw_lock_win.h b/third_party/libwebrtc/webrtc/system_wrappers/source/rw_lock_win.h
index 0e5ecf160394..41537ba10ba5 100644
--- a/third_party/libwebrtc/webrtc/system_wrappers/source/rw_lock_win.h
+++ b/third_party/libwebrtc/webrtc/system_wrappers/source/rw_lock_win.h
@@ -13,7 +13,7 @@
 
 #include "system_wrappers/include/rw_lock_wrapper.h"
 
-#include <windows.h>
+#include <Windows.h>
 
 namespace webrtc {
 
diff --git a/third_party/libwebrtc/webrtc/test/win/d3d_renderer.h b/third_party/libwebrtc/webrtc/test/win/d3d_renderer.h
index a5a8506789da..68a5a802f2ee 100644
--- a/third_party/libwebrtc/webrtc/test/win/d3d_renderer.h
+++ b/third_party/libwebrtc/webrtc/test/win/d3d_renderer.h
@@ -10,7 +10,7 @@
 #ifndef TEST_WIN_D3D_RENDERER_H_
 #define TEST_WIN_D3D_RENDERER_H_
 
-#include <windows.h>
+#include <Windows.h>
 #include <d3d9.h>
 #pragma comment(lib, "d3d9.lib")       // located in DirectX SDK
 
diff --git a/third_party/libwebrtc/webrtc/test/win/run_loop_win.cc b/third_party/libwebrtc/webrtc/test/win/run_loop_win.cc
index 1b5c3bf1c16b..494adc47d60d 100644
--- a/third_party/libwebrtc/webrtc/test/win/run_loop_win.cc
+++ b/third_party/libwebrtc/webrtc/test/win/run_loop_win.cc
@@ -13,7 +13,7 @@
 
 #include <conio.h>
 #include <stdio.h>
-#include <windows.h>
+#include <Windows.h>
 
 namespace webrtc {
 namespace test {
diff --git a/third_party/libwebrtc/webrtc/webrtc_gn/moz.build b/third_party/libwebrtc/webrtc/webrtc_gn/moz.build
index 98889f7ce5d7..7ce0675b163b 100644
--- a/third_party/libwebrtc/webrtc/webrtc_gn/moz.build
+++ b/third_party/libwebrtc/webrtc/webrtc_gn/moz.build
@@ -217,7 +217,7 @@ if CONFIG["OS_TARGET"] == "WINNT":
     OS_LIBS += [
         "d3d11",
         "dxgi",
-        "strmiids",
+        "Strmiids",
         "winmm"
     ]
 
diff --git a/third_party/sipcc/cpr_win_types.h b/third_party/sipcc/cpr_win_types.h
index 1bcdc75a7b90..c4dfa0b72a49 100644
--- a/third_party/sipcc/cpr_win_types.h
+++ b/third_party/sipcc/cpr_win_types.h
@@ -40,19 +40,15 @@ typedef uint8_t boolean;
  * NOTE: size_t should already be declared by both the MinGW and Microsoft
  * SDKs.
  */
-#ifndef __MINGW32__
 #ifndef _SSIZE_T_
 #define _SSIZE_T_
 typedef int ssize_t;
 #endif
-#endif
 
 /*
  * Define pid_t.
  */
-#ifndef __MINGW32__
 typedef int pid_t;
-#endif
 
 /*
  * Define min/max
diff --git a/third_party/sipcc/moz.build b/third_party/sipcc/moz.build
index d5a179403bfd..851032c37020 100644
--- a/third_party/sipcc/moz.build
+++ b/third_party/sipcc/moz.build
@@ -6,7 +6,6 @@
 if CONFIG['MOZ_WEBRTC_SIGNALING']:
     if CONFIG['OS_TARGET'] == 'WINNT':
         DEFINES['SIP_OS_WINDOWS'] = True
-        DEFINES['WEBRTC_MOZILLA_BUILD'] = 1
         SOURCES += [
             'sdp_services_win32.c',
         ]
diff --git a/third_party/sipcc/sdp_token.c b/third_party/sipcc/sdp_token.c
index bb35faa21ea3..b570d8169009 100644
--- a/third_party/sipcc/sdp_token.c
+++ b/third_party/sipcc/sdp_token.c
@@ -11,10 +11,6 @@
 #include "sdp_log.h"
 #include "prprf.h"
 
-#if __MINGW32__
-#include <minmax.h>
-#endif
-
 static const char *logTag = "sdp_token";
 
 #define MCAST_STRING_LEN 4
diff --git a/toolkit/content/gmp-sources/widevinecdm.json b/toolkit/content/gmp-sources/widevinecdm.json
index 5246e9ce737b..e3cb9c0ad476 100644
--- a/toolkit/content/gmp-sources/widevinecdm.json
+++ b/toolkit/content/gmp-sources/widevinecdm.json
@@ -47,9 +47,6 @@
         },
         "WINNT_x86_64-msvc-x64": {
           "alias": "WINNT_x86_64-msvc"
-        },
-        "WINNT_x86_64-gcc3-x64": {
-          "alias": "WINNT_x86_64-msvc"
         }
       },
       "version": "4.10.2209.1"
-- 
2.34.0

